### PR TITLE
TEST: Add limit to number of ignored error messages

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: test/gtest/common/test.cc [hunk 1]
```diff
@@ -146,6 +146,16 @@ std::string test_base::format_message(const char *message, va_list ap)
     return std::string(buf);
 }
 
+void test_base::push_debug_message_with_limit(std::vector<std::string>& vec,
+                                              const std::string& message,
+                                              const size_t limit) {
+    if (vec.size() >= limit) {
+        UCS_TEST_ABORT("too many errors/warnings; bailing out (" + message + ")");
```
Comment (alinask) [inline test/gtest/common/test.cc]: maybe print the limit?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9703/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9708/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9734/ for details (Mellanox internal link).

**Pull request merged**